src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -65,10 +68,28 @@ class DependencyTools(object):
         creating error and warning messages.
     :type language_writer: None (default is Fortran), or an instance \
         of :py:class:`psyclone.psyir.backend.visitor.PSyIRVisitor`
+
+    :raises TypeError: if an invalid loop type is specified.
+
     '''
     def __init__(self, loop_types_to_parallelise=None,
                  language_writer=FortranWriter()):
         if loop_types_to_parallelise:
+            # Verify that all loop types specified are valid:
+            config = Config.get()
+            constants = config.get_constants()
+            for loop_type in loop_types_to_parallelise:
+                if loop_type not in constants.VALID_LOOP_TYPES:
+                    if six.PY2:
+                        out_list = [str(l_type)for l_type in
```

Comment:
missing space before "for"

Reply:
Strangely again my pylint did not detect this :(

Reply:
I dub myself "Andy Pylint Porter" :-)